include/graybat/Cage.hpp

```diff
@@ -683,6 +692,29 @@ namespace graybat {
     template<typename T_CommunicationPolicy, typename T_GraphPolicy>
     template<typename T>
     auto Cage<T_CommunicationPolicy, T_GraphPolicy>::
+	recv (const Edge &edge, std::future<T>& data)
+	-> void
+	{
+		VAddr srcVAddr = locateVertex(edge.source);
+		std::promise<T> dataPromise;
+		data = dataPromise.get_future();
+
+		// The reason for the std::promise and std::thread over std::async ist the destructor behaviour of std::async
+		// For std::async the destructor is waiting for the completion of the asyncThread, which may lead to a permanent lock
+		// The future from std::promise does not block on the completion of the execution thread.
+		std::thread asyncRecvThread([&](){
+			T result;
+			Event e = comm->asyncRecv(srcVAddr, edge.id, graphContext, result);
+			e.wait();
+			dataPromise.set_value(result);
+		});
+		asyncRecvThread.detach();
+	}
```

Comment:
My problem was, that the event class has no ```wait_for(duration)```. The purpose of the Event class is to replicate the std::future behaviour and i thought it would be a nicer and cleaner way to get rid of the event class.

For the interface, i think a nice solution would be:
```C++
auto recvData = cage.recv<DataType>(edge); // Return value, blocking
auto asyncData = cage.asyncRecv<DataType>(edge); // Return std::future
```

I get the point with the unbound threading. Would a threadpool, that is bound to the existance of the cage solve the problem?